src/list.js

```diff
@@ -48,17 +48,25 @@ class List extends React.Component {
     };
 
     this.handleClick = this.handleClick.bind(this);
+    this.clearChecklistAndLocalStorage = this.clearChecklistAndLocalStorage.bind(this);
   }
 
   handleClick(relicId) {
     let checklist = this.state.checklist;
-    if (!this.state.checklist.includes(relicId)) checklist.push(relicId);
+    if (!checklist.includes(relicId)) checklist.push(relicId);
     else checklist.splice(checklist.indexOf(relicId), 1);
     this.props.onChangeItems(checklist);
     localStorage.setItem("checklist", JSON.stringify(checklist));
     this.setState({checklist: checklist});
   }
 
+  clearChecklistAndLocalStorage() {
```

Comment:
I'd create separate methods or call this one something unified e.g. `clearState`.